src/js/actions/authActions.js

```diff
@@ -0,0 +1,85 @@
+import fetch from 'fetch-everywhere';
+import {push} from 'react-router-redux';
+import urlFor from '../helpers/urlFor';
+
+const requestLogout = () => ({
+  type: 'LOGOUT_REQUEST',
+  isFetching: true,
+  isAuthenticated: true
+});
+
+const receiveLogout = () => ({
+  type: 'LOGOUT_SUCCESS',
+  isFetching: false,
+  isAuthenticated: false
+});
+
+// Logs the user out
+export const logoutUser = () => (dispatch) => {
+  dispatch(requestLogout());
+  localStorage.removeItem('idToken');
+  localStorage.removeItem('store');
+  dispatch(receiveLogout());
+  dispatch(push('/'));
+};
+
+const requestLogin = (code) => ({
+  type: 'LOGIN_REQUEST',
+  isFetching: true,
+  isAuthenticated: false,
+  code
+});
+
+const receiveLogin = (token) => ({
+  type: 'LOGIN_SUCCESS',
+  isFetching: false,
+  isAuthenticated: true,
+  user: token
+});
+
+const loginError = (err) => ({
+  type: 'LOGIN_FAILURE',
+  isFetching: false,
+  isAuthenticated: false,
+  err
+});
+
+export const loginUser = (code) => (dispatch) => {
+  // API Route for Slack Authentication
+  const url = urlFor('/api/v1/auth/slack');
```

Comment:
Interesting - I actually knew about the scoping issue with var, just didn't know there was a way around it. It's always just been one of those quirks of the language that you learn to deal with. A quick look at the browser compatibility chart for ES6 shows it's actually becoming viable to use too, neat!

My note about all cap constant names still applies though right?